gitclone/steperror.go

```diff
@@ -1,11 +1,178 @@
 package gitclone
 
-import "github.com/bitrise-io/bitrise-init/step"
+import (
```

Comment:
I suggest splitting this file into two: 
- error_matcher.go (includes `PatternErrorMatcher ` and related types `DetailBuilder `, `Detail `, ..)
- steperror.go (includes `mapRecommendation`, step specific implementations)

so that later we can easily move the shared code to a new package.